src/ReverseProxy/Service/RuntimeModel/Transforms/StructuredTransformer.cs

```diff
@@ -20,192 +20,137 @@ namespace Microsoft.ReverseProxy.Service.RuntimeModel.Transforms
     /// </summary>
     internal class StructuredTransformer : HttpTransformer
     {
-        private static readonly HashSet<string> EmptyHash = new HashSet<string>(0);
-
         /// <summary>
         /// Creates a new <see cref="StructuredTransformer"/> instance.
         /// </summary>
-        internal StructuredTransformer(bool? copyRequestHeaders, IList<RequestParametersTransform> requestTransforms,
-            Dictionary<string, RequestHeaderTransform> requestHeaderTransforms,
-            Dictionary<string, ResponseHeaderTransform> responseHeaderTransforms,
-            Dictionary<string, ResponseHeaderTransform> responseTrailerTransforms)
+        internal StructuredTransformer(bool? copyRequestHeaders, bool? copyResponseHeaders, bool? copyResponseTrailers,
+            IList<RequestTransform> requestTransforms,
+            IList<ResponseTransform> responseTransforms,
+            IList<ResponseTrailersTransform> responseTrailerTransforms)
         {
             ShouldCopyRequestHeaders = copyRequestHeaders;
+            ShouldCopyResponseHeaders = copyResponseHeaders;
```

Comment:
We could set the value in constructor only once and change the `bool?` to `bool`.
```suggestion
            ShouldCopyResponseHeaders = copyResponseHeaders.GetValueOrDefault(true);
```

Reply:
I'd left it this way so I could test TransformBuilder for the cases where no config was provided vs an explicit negative config given. If it shows up in a trace then I'll remove it and find another way to test the builder.